mesosphere/kudo-spark-operator
# DCOS-59602 [Spark Operator] Add CRDss templates to kudo-spark-operator definition

### What changes were proposed in this pull request?
1) Resolves [DCOS-59602: Add CRDs templates to kudo-spark-operator definition](https://jira.mesosphere.com/browse/DCOS-59602)
1) Upgrade to `KUDO 0.7.3` (it's vital to have ability install CRDs within the operator). If you would like better understand the topic, please, take a look on the [message](https://mesosphere.slack.com/archives/CKKDWQDGW/p1570457285154700) and the [thread](https://mesosphere.slack.com/archives/CKKDWQDGW/p1570457499156000) in Slack.
1) Fixes `remove_operator.py`: 
   - Removed deletion of `planexecutions.kudo.dev`, because that Kind no longer exists with KUDO 0.7.3;
   - Fix ignoring specified `NAMESPACE` during resources deletion.

### Why are the changes needed?
Having `kudo-spark-operator` installed, these changes introduce ability to deploy Spark jobs using `SparkApplication` and `ScheduledSparkApplication` kinds, defined in CRDs.

### How were the changes tested?
#### Prerequisite
- Konvoy cluster up and running.
- kubectl.
- KUDO-cli **v0.7.3**.
- Python3.
**_if you would like use cluster with any previous version of KUDO installed, then you need delete KUDO version beforehand._**
#### How to remove KUDO from cluster? 
1) `kubectl kudo init --dry-run -o yaml | kubectl delete -f -`
2) Delete all KUDO's CRDs - `kubectl delete crd *.kudo.dev`
3) `kubectl kudo init -wait`

#### How to verify the PR changes
1) Install/upgrade `kudo-cli` to [**0.7.3 version**](https://github.com/kudobuilder/kudo/releases/tag/v0.7.3). **_Critically important! [here](https://mesosphere.slack.com/archives/CKKDWQDGW/p1570457285154700) is why_**
1) Install `kudo-spark-operator` with via `NAMESPACE=<desired_namespace>  ./scripts/install_operator.sh`
1) Check that CRDs are installed - `kubectl get crd | grep spark`. Output should be like this - 
```bash
$ kubectl get crd | grep spark
scheduledsparkapplications.sparkoperator.k8s.io   2019-10-08T14:03:52Z
sparkapplications.sparkoperator.k8s.io            2019-10-08T14:03:53Z
```
1) Look into Kubernetes Dashboard, waiting Spark driver is up and running (green).
1) Deploy `SparkApplication` example -
`kubectl -n <desired_namepace> apply -f ./specs/spark-application.yaml`
1) Look into Kubernetes Dashboard, waiting Spark Job is up and running (green).
1) Delete operator `NAMESPACE=<>desired_namespace> ./scripts/remove_operator.py`
1) Check that CRDs are removed from cluster now - `kubectl get crd | grep spark`
